### PR TITLE
Update Kubernetes v1.25 conformance test coverage

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
Updates Kubernetes conformance test coverage for v1.25.
Ref: https://github.com/kubernetes/test-infra/pull/28186

**Which issue(s) this PR fixes**:
Part of #6567 

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,11 +40,11 @@ Continuous conformance test results of the latest stable Gardener release are up
...
...ance-gardener/Gardener,%20v1.25%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.25%20AWS)  | [![Gardener v1.24 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.24%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.24%20AWS) | [![Gardener v1.23 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.23%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.23%20AWS) | [![Gardener v1.22 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.22%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.22%20AWS) | [![Gardener v1.21 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.21%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.21%20AWS) | [![Gardener v1.20 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.20%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.20%20AWS) |
```
Comment (ialidzhikov) [inline README.md]: Cosmetic nit: the aws and alicloud links have 2 spaces around the 1.25 link. Can we reduce it to 1 space for consistency reasons with the other links?

**Review by ialidzhikov — COMMENTED** [review]
@shafeeqes , thanks for the PR! I assume we can now promote it ready? At least the tiles are visible to me, if that was the reason to create the PR as draft.

**Comment by shafeeqes** [comment]
> @shafeeqes , thanks for the PR! I assume we can now promote it ready? At least the tiles are visible to me, if that was the reason to create the PR as draft.

Yeah, I was wondering if we should wait till we can update 1.25 here as well,
https://github.com/gardener/gardener/blob/5a69fa603d12ed6a31424bb7e4adbd25b5c73a08/README.md#L37

**Comment by ialidzhikov** [comment]
I don't know what is the process of updating this spreadsheet. IMO you can update the text that K8s versions up to v1.25 are supported. We can proceed with this PR and then try follow up internally on how edits were performed to this spreadsheet in the past.

**Comment by shafeeqes** [comment]
> I don't know what is the process of updating this spreadsheet. IMO you can update the text that K8s versions up to v1.25 are supported. We can proceed with this PR and then try follow up internally on how edits were performed to this spreadsheet in the past.

Sure.

**Comment by ialidzhikov** [comment]
Thanks. Can we also update the text that we discussed above and the link to this spreadsheet (as we are anyways touching this part)?

```diff
-Gardener takes part in the [Certified Kubernetes Conformance Program](https://www.cncf.io/certification/software-conformance/) to attest its compatibility with the K8s conformance testsuite. Currently Gardener is certified for K8s versions up to v1.24, see [the conformance spreadsheet](https://docs.google.com/spreadsheets/d/1uF9BoDzzisHSQemXHIKegMhuythuq_GL3N1mlUUK2h0/edit#gid=0&range=121:122).
+Gardener takes part in the [Certified Kubernetes Conformance Program](https://www.cncf.io/certification/software-conformance/) to attest its compatibility with the K8s conformance testsuite. Currently Gardener is certified for K8s versions up to v1.25, see [the conformance spreadsheet](https://docs.google.com/spreadsheets/d/1uF9BoDzzisHSQemXHIKegMhuythuq_GL3N1mlUUK2h0/edit#gid=0&range=126:127).
```

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/approve

**Comment by ialidzhikov** [comment]
@shafeeqes can you also prepare PRs for the extensions?

**Comment by shafeeqes** [comment]
> @shafeeqes can you also prepare PRs for the extensions?

Sure, Will do it early next week.

**Pull request merged**